calico_node/tests/st/policy/test_felix_gateway.py

```diff
@@ -844,7 +844,12 @@ def delete_all(self, resource):
             if 'items' in objects and len(objects['items']) == 0:
                 pass
             else:
-                self._delete_data(objects, self.hosts[0])
+                # clean out the incorrectly formatted python datetime timestamps
+                if 'items' in objects:
+                    for obj in objects['items']:
+                        if 'metadata' in obj and isinstance(obj['metadata'], dict):
+                            obj['metadata'].pop('creationTimestamp')
+                    self._delete_data(objects, self.hosts[0])
```

Comment:
this should not be under the if statement, we should delete_data irrespective of items in objects ....